Bugzilla/WebService/User.pm

```diff
@@ -183,23 +183,32 @@ sub suggest {
   }
   $where = "($where) AND is_enabled = 1";
 
-  my $sql
-    = 'SELECT '
-    . join(', ', @select)
-    . " FROM profiles WHERE $where ORDER BY $order LIMIT 25";
-  my $results = $dbh->selectall_arrayref($sql, {Slice => {}});
+  my $results = $dbh->selectall_arrayref(
+    "SELECT "
+      . join(', ', @select)
+      . " FROM profiles WHERE $where ORDER BY $order LIMIT 25",
+    {Slice => {}}
+  );
+  my $user_objects = Bugzilla::User->new_from_list([map { $_->{id} } @$results]);
```

Comment:
When originally authored there was a reasonable performance hit using the User objects for the number of objects we create.  Please ensure you don't regress performance.

Reply:
Interestingly, this does not impact performance from what we have currently.  Reason being is that the extensions hooks for the Review and Gravatar extensions already loaded full user objects for the entire list of user ids in order to add their data to the results. This was before bug 1787954. So the changes are just doing it earlier in the code path. 

That being said, we could definitely make this perform better than it does currently as a different bug and also add some testing for performance regressions which we do not have at the moment for any aspect of BMO afaict.

Reply:
Thanks - makes sense.